### ISSUE TITLE
install amqp && pecl info amqp returned 1 instead of one of [0]

### ISSUE THREAD

**Issue by stanislas-prime** [puphpet/puphpet#2105]
Hi, I'm trying to install php 5.6 and RabbitMQ. During vagrant up or provision get this message: "install amqp && pecl info amqp returned 1 instead of one of [0]". It tells me about failed installation of dependecy for php library for working with RabbitMQ.
```
==> local: Running provisioner: shell...
    local: Running: inline script
==> local: Running provisioner: shell...
    local: Running: C:/Users/stani/AppData/Local/Temp/vagrant-shell20160116-7940-w3o755.sh
==> local: 
==> local:  ____        ____  _   _ ____      _      generated using
==> local: |  _ \ _   _|  _ \| | | |  _ \ ___| |_   ___ ___  _ __ ___
==> local: | |_) | | | | |_) | |_| | |_) / _ \ __| / __/ _ \| '_ ` _ \
==> local: |  __/| |_| |  __/|  _  |  __/  __/ |_ | (_| (_) | | | | | |
==> local: |_|    \__,_|_|   |_| |_|_|   \___|\__(_)___\___/|_| |_| |_|
==> local: Running provisioner: shell...
    local: Running: C:/Users/stani/AppData/Local/Temp/vagrant-shell20160116-7940-o3z5lf.sh
==> local: Pre-existing private key found at 'puphpet/files/dot/ssh/root_id_rsa'
==> local: Pre-existing private key found at 'puphpet/files/dot/ssh/id_rsa'
==> local: Adding generated root key to /root/.ssh/id_rsa
==> local: Adding generated root key to /root/.ssh/id_rsa.pub
==> local: Adding generated root key to /root/.ssh/authorized_keys
==> local: Adding generated key to /home/vagrant/.ssh/id_rsa
==> local: Adding generated key to /home/vagrant/.ssh/id_rsa.pub
==> local: Adding generated key to /home/vagrant/.ssh/authorized_keys
==> local: Running provisioner: shell...
    local: Running: C:/Users/stani/AppData/Local/Temp/vagrant-shell20160116-7940-1l93v6w.sh
==> local: Running provisioner: shell...
    local: Running: C:/Users/stani/AppData/Local/Temp/vagrant-shell20160116-7940-124sxxr.sh
==> local: Successfully installed deep_merge-1.0.1
==> local: Successfully installed i18n-0.7.0
==> local: Successfully installed minitest-5.8.3
==> local: Successfully installed thread_safe-0.3.5
==> local: Successfully installed tzinfo-1.2.2
==> local: Successfully installed activesupport-4.2.5
==> local: Successfully installed vine-0.2
==> local: 7 gems installed
==> local: Running provisioner: shell...
    local: Running: C:/Users/stani/AppData/Local/Temp/vagrant-shell20160116-7940-1e03li9.sh
==> local: Running files in files/exec-preprovision
==> local: Finished running files in files/exec-preprovision
==> local: To run again, delete hashes you want rerun in /.puphpet-stuff/exec-preprovision-ran or the whole file to rerun all
==> local: Running files in files/
==> local: Finished running files in files/
==> local: Running provisioner: puppet...
==> local: Running Puppet with ...
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Info: Loading facts
==> local: Warning: Scope(Apt::Source[nginx]): $key_source is deprecated and will be removed in the next major release, please use $key => { 'source' => http://nginx.org/keys/nginx_signing.key } instead.
==> local: Warning: Scope(Apt::Source[rabbitmq]): $include_src is deprecated and will be removed in the next major release, please use $include => { 'src' => false } instead
==> local: Warning: Scope(Apt::Source[rabbitmq]): $key_source is deprecated and will be removed in the next major release, please use $key => { 'source' => http://www.rabbitmq.com/rabbitmq-signing-key-public.asc } instead.
==> local: Warning: Scope(Apt::Key[Add key: 7BD9BF62 from Apt::Source nginx]): $key_source is deprecated and will be removed in the next major release. Please use $source instead.
==> local: Warning: Scope(Apt::Key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq]): $key_source is deprecated and will be removed in the next major release. Please use $source instead.
==> local: Notice: Compiled catalog for rabbitmq in environment production in 6.21 seconds
==> local: Warning: Found multiple default providers for vcsrepo: bzr, cvs, dummy, git, hg, svn; using bzr
==> local: Warning: /Apt_key[Add key: 7BD9BF62 from Apt::Source nginx]: The id should be a full fingerprint (40 characters), see README.
==> local: Info: Applying configuration version '1452966264'
==> local: Notice: /Stage[main]/Puphpet_php/Exec[mkdir -p /var/lib/php/session]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Exec[mkdir -p /var/lib/php/session]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_nginx/File[/var/www]/owner: owner changed 'www-data' to 'root'
==> local: Notice: /Stage[main]/Puphpet_php/File[/var/lib/php/session]/owner: owner changed 'root' to 'www-data'
==> local: Notice: /Stage[main]/Puphpet_php/File[/var/lib/php/session]/group: group changed 'root' to 'www-data'
==> local: Notice: /Stage[main]/Puphpet_server/Exec[dotfiles]/returns: executed successfully
==> local: Notice: /Stage[main]/Puphpet::Php::Repos/Apt::Ppa[ppa:ondrej/php5-5.6]/Exec[add-apt-repository-ppa:ondrej/php5-5.6]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet::Php::Repos/Apt::Ppa[ppa:ondrej/php5-5.6]/Exec[add-apt-repository-ppa:ondrej/php5-5.6]: Scheduling refresh of Class[Apt::Update]
==> local: Info: Class[Apt::Update]: Scheduling refresh of Exec[apt_update]
==> local: Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
==> local: Notice: /Stage[main]/Php/Package[php5-fpm]/ensure: created
==> local: Notice: /Stage[main]/Php/File[php.conf]/ensure: created
==> local: Notice: /Stage[main]/Php::Devel/Package[php5-dev]/ensure: created
==> local: Notice: /Stage[main]/Puphpet_php/Exec[set php session path owner/group]/returns: executed successfully
==> local: Notice: /Stage[main]/Composer/Exec[download_composer]/returns: executed successfully
==> local: Notice: /Stage[main]/Composer/File[/usr/local/bin/composer]/ensure: defined content as '{md5}26090dfc90836ff664a091c371895f5b'
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Ini[error_log: /var/log/php-fpm.log]/Augeas[global/error_log: /var/log/php-fpm.log]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Ini[error_log: /var/log/php-fpm.log]/Augeas[global/error_log: /var/log/php-fpm.log]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Ini[pid: /run/php-fpm.pid]/Augeas[global/pid: /run/php-fpm.pid]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Ini[pid: /run/php-fpm.pid]/Augeas[global/pid: /run/php-fpm.pid]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Pool_ini[www/listen: 127.0.0.1:9000]/Augeas[www/listen: 127.0.0.1:9000]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Pool_ini[www/listen: 127.0.0.1:9000]/Augeas[www/listen: 127.0.0.1:9000]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Pool_ini[www/security.limit_extensions: .php]/Augeas[www/security.limit_extensions: .php]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Pool_ini[www/security.limit_extensions: .php]/Augeas[www/security.limit_extensions: .php]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Pool_ini[www/user: www-user]/Augeas[www/user: www-user]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Fpm::Pool_ini[www/user: www-user]/Augeas[www/user: www-user]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[cgi.fix_pathinfo]/File[/etc/php5/mods-available/zzzz_custom.ini]/ensure: created
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[cgi.fix_pathinfo]/File[/etc/php5/conf.d]/ensure: created
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[cgi.fix_pathinfo]/File[/etc/php5/fpm/conf.d/zzzz_custom.ini]/ensure: created
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[cgi.fix_pathinfo]/File[/etc/php5/cli/conf.d/zzzz_custom.ini]/ensure: created
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[cgi.fix_pathinfo]/Augeas[CUSTOM/cgi.fix_pathinfo: 1]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[cgi.fix_pathinfo]/Augeas[CUSTOM/cgi.fix_pathinfo: 1]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[display_errors]/Augeas[CUSTOM/display_errors: On]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[display_errors]/Augeas[CUSTOM/display_errors: On]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[error_reporting]/Augeas[CUSTOM/error_reporting: -1]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[error_reporting]/Augeas[CUSTOM/error_reporting: -1]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[session.save_path]/Augeas[CUSTOM/session.save_path: /var/lib/php/session]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[session.save_path]/Augeas[CUSTOM/session.save_path: /var/lib/php/session]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[date.timezone]/Augeas[CUSTOM/date.timezone: UTC]/returns: executed successfully
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Ini[date.timezone]/Augeas[CUSTOM/date.timezone: UTC]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Pyenv/Vcsrepo[/usr/local/pyenv]/revision: revision changed '3cde540ca00208fbfd0a5e7708c3053bc8cca9f5' to 'v0.4.0-20140404'
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Module[intl]/Php::Module[intl]/Package[PhpModule_intl]/ensure: created
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Module[intl]/Php::Module[intl]/Package[PhpModule_intl]: Scheduling refresh of Service[php5-fpm]
==> local: Info: Puphpet::Php::Module[intl]: Scheduling refresh of Service[php5-fpm]
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: downloading amqp-1.6.1.tgz ...
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: Starting to download amqp-1.6.1.tgz (61,998 bytes)
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: ................done: 61,998 bytes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: 18 source files, building
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: running: phpize
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: Configuring for:
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: PHP Api Version:         20131106
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: Zend Module Api No:      20131226
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: Zend Extension Api No:   220131226
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: building in /tmp/pear/temp/pear-build-defaultuserMJPDGs/amqp-1.6.1
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: running: /tmp/pear/temp/amqp/configure
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for grep that handles long lines and -e... /bin/grep
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for egrep... /bin/grep -E
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for a sed that does not truncate output... /bin/sed
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for cc... cc
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking whether the C compiler works... yes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for C compiler default output file name... a.out
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for suffix of executables...
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking whether we are cross compiling... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for suffix of object files... o
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking whether we are using the GNU C compiler... yes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking whether cc accepts -g... yes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for cc option to accept ISO C89... none needed
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking how to run the C preprocessor... cc -E
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for icc... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for suncc... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking whether cc understands -c and -o together... yes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for system library directory... lib
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking if compiler supports -R... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking if compiler supports -Wl,-rpath,... yes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking build system type... x86_64-unknown-linux-gnu
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking host system type... x86_64-unknown-linux-gnu
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking target system type... x86_64-unknown-linux-gnu
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for PHP prefix... /usr
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for PHP includes... -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for PHP extension directory... /usr/lib/php5/20131226
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for PHP installed headers prefix... /usr/include/php5
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking if debug is enabled... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking if zts is enabled... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for re2c... no
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for gawk... gawk
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for amqp support... yes, shared
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for amqp... yes, shared
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: yes
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: checking for amqp files in default path... not found
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: configure: error: Please reinstall the librabbitmq distribution itself or (re)install librabbitmq development package if it available in your system
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: ERROR: `/tmp/pear/temp/amqp/configure' failed
==> local: Notice: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: No information found for `amqp'
==> local: Error: printf "\n" | pecl -d preferred_state=stable install amqp && pecl info amqp returned 1 instead of one of [0]
==> local: Error: /Stage[main]/Puphpet_rabbitmq/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: change from notrun to 0 failed: printf "\n" | pecl -d preferred_state=stable install amqp && pecl info amqp returned 1 instead of one of [0]
==> local: Notice: /Stage[main]/Puphpet_php/Service[php5-fpm]: Dependency Exec[pecl-amqp] has failures: true
==> local: Warning: /Stage[main]/Puphpet_php/Service[php5-fpm]: Skipping because of failed dependencies
==> local: Info: /Stage[main]/Puphpet_php/Service[php5-fpm]: Unscheduling all events on Service[php5-fpm]
==> local: Info: Class[Puphpet_php]: Unscheduling all events on Class[Puphpet_php]
==> local: Info: Stage[main]: Unscheduling all events on Stage[main]
==> local: Notice: Applied catalog in 84.30 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
If I login to server via ssh and try to reinstall amqp throught pecl get this `sudo pecl install amqp`:
```
Starting to download amqp-1.6.1.tgz (61,998 bytes)
................done: 61,998 bytes
18 source files, building
running: phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226
building in /tmp/pear/temp/pear-build-rootAvukIj/amqp-1.6.1
running: /tmp/pear/temp/amqp/configure
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib
checking for PHP extension directory... /usr/lib/php5/20131226
checking for PHP installed headers prefix... /usr/include/php5
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking for amqp support... yes, shared
checking for amqp... yes, shared
yes
checking for amqp files in default path... not found
configure: error: Please reinstall the librabbitmq distribution itself or (re)install librabbitmq development package if it available in your system
ERROR: `/tmp/pear/temp/amqp/configure' failed
```
Ok... Then I am trying to install librabbitmq development package:
`sudo apt-get install librabbitmq-dev`
```
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following extra packages will be installed:
  librabbitmq1
The following NEW packages will be installed:
  librabbitmq-dev librabbitmq1
0 upgraded, 2 newly installed, 0 to remove and 43 not upgraded.
Need to get 62.0 kB of archives.
After this operation, 280 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://us.archive.ubuntu.com/ubuntu/ trusty/main librabbitmq1 amd64 0.4.1-1 [35.2 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu/ trusty/main librabbitmq-dev amd64 0.4.1-1 [26.9 kB]
Fetched 62.0 kB in 0s (66.6 kB/s)
Selecting previously unselected package librabbitmq1.
(Reading database ... 73073 files and directories currently installed.)
Preparing to unpack .../librabbitmq1_0.4.1-1_amd64.deb ...
Unpacking librabbitmq1 (0.4.1-1) ...
Selecting previously unselected package librabbitmq-dev.
Preparing to unpack .../librabbitmq-dev_0.4.1-1_amd64.deb ...
Unpacking librabbitmq-dev (0.4.1-1) ...
Setting up librabbitmq1 (0.4.1-1) ...
Setting up librabbitmq-dev (0.4.1-1) ...
Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
```
Ok, trying to install pecl amqp once again:
`sudo pecl install amqp`
```
downloading amqp-1.6.1.tgz ...
Starting to download amqp-1.6.1.tgz (61,998 bytes)
................done: 61,998 bytes
18 source files, building
running: phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226
building in /tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1
running: /tmp/pear/temp/amqp/configure
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib
checking for PHP extension directory... /usr/lib/php5/20131226
checking for PHP installed headers prefix... /usr/include/php5
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking for amqp support... yes, shared
checking for amqp... yes, shared
yes
checking for amqp files in default path... found in /usr
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... (cached) gawk
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating ./config.status
config.status: creating config.h
config.status: executing libtool commands
running: make
/bin/bash /tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/libtool --mode=compile cc  -I. -I/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/include -I/tmp/pe
ar/temp/pear-build-rooteCZHWZ/amqp-1.6.1/main -I/tmp/pear/temp/amqp -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr
/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/pear/temp/amqp/amqp.c -o amqp.lo
libtool: compile:  cc -I. -I/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/include -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/main -I/tmp/pear/temp/
amqp -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/
temp/amqp/amqp.c  -fPIC -DPIC -o .libs/amqp.o
/bin/bash /tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/libtool --mode=compile cc  -I. -I/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/include -I/tmp/pe
ar/temp/pear-build-rooteCZHWZ/amqp-1.6.1/main -I/tmp/pear/temp/amqp -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr
/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/pear/temp/amqp/amqp_exchange.c -o amqp_exchange.lo
libtool: compile:  cc -I. -I/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/include -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/main -I/tmp/pear/temp/
amqp -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/
temp/amqp/amqp_exchange.c  -fPIC -DPIC -o .libs/amqp_exchange.o
/bin/bash /tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/libtool --mode=compile cc  -I. -I/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/include -I/tmp/pe
ar/temp/pear-build-rooteCZHWZ/amqp-1.6.1/main -I/tmp/pear/temp/amqp -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr
/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/pear/temp/amqp/amqp_queue.c -o amqp_queue.lo
libtool: compile:  cc -I. -I/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/include -I/tmp/pear/temp/pear-build-rooteCZHWZ/amqp-1.6.1/main -I/tmp/pear/temp/
amqp -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/
temp/amqp/amqp_queue.c  -fPIC -DPIC -o .libs/amqp_queue.o
In file included from /tmp/pear/temp/amqp/amqp_queue.c:51:0:
/tmp/pear/temp/amqp/amqp_queue.c: In function 'convert_amqp_envelope_to_zval':
/tmp/pear/temp/amqp/amqp_queue.c:299:51: error: 'AMQP_DELIVERY_NONPERSISTENT' undeclared (first use in this function)
   AMQP_SET_LONG_PROPERTY(envelope->delivery_mode, AMQP_DELIVERY_NONPERSISTENT);
                                                   ^
/tmp/pear/temp/amqp/php_amqp.h:253:14: note: in definition of macro 'AMQP_SET_LONG_PROPERTY'
  (object) = (value);
              ^
/tmp/pear/temp/amqp/amqp_queue.c:299:51: note: each undeclared identifier is reported only once for each function it appears in
   AMQP_SET_LONG_PROPERTY(envelope->delivery_mode, AMQP_DELIVERY_NONPERSISTENT);
                                                   ^
/tmp/pear/temp/amqp/php_amqp.h:253:14: note: in definition of macro 'AMQP_SET_LONG_PROPERTY'
  (object) = (value);
              ^
make: *** [amqp_queue.lo] Error 1
ERROR: `make' failed
```
So?... Please help me to resolve this issue and install pecl package for amqp.

**Comment by stanislas-prime** [comment]
There is an error during the installation PHP extension for RabbitMQ. The thing is that system tries to install `librabbitmq 0.4.1-1`, which it too old for `php-amqp 1.6.1`. From [php-amqp Requirements](https://github.com/pdezwart/php-amqp#requirements):
> since php-amqp>=1.6.0 librabbitmq >= 0.5.2 required, >= 0.6.0 recommended

So... puppet model need to be updated to install latest librabbitmq-dev package.

Also there is a problem with accessign RabbitMQ management page in browser. I recognized that to solve the problem we need to add exception to `iptables` for port `15672`:
`iptables -I INPUT -p tcp --dport 15672 -j ACCEPT`

After that management page is accessible. Will be greate if all these things will be fixed. Thanks.

**Comment by jtreminio** [comment]
Closing in favor of #2117
